File: Python/src/gestorAplicacion/gestion/Persona.py
# /*  Autores: Joan Sebastian Salazar Montoya, Monica Sofia Restrepo León, 
# Luis Alejandro Varela Ojeda, Maria Fernanda Calle Agudelo, Jaider Castañeda Villa
#  * FUNCIONALIDAD DE LA CLASE: clase abstracta que proporciona atributos y métodos 
#  * comunes para la creación de los distintos tipos de trabajadores de la empresa.
#  */

class Persona:
    #/*-------------------Atributos-------------------------*/
    personasTotales = 0
    listaPersonas = []
    SALARIO=1000
    #/*-------------------Constructores-------------------------*/
    def __init__(self, nombre, edad, cedula, cuentaBancaria):
        self._nombre = nombre
        self._edad = edad
        self._cedula = cedula
        self._cuentaBancaria = cuentaBancaria
        self._trabajo = 0
        self._indiceMeta = 0
        self._verificadorMetasCumplidas = [False, False]
        Persona.listaPersonas.append(self)
    #/*------------------- Metodos -------------------------*/
    def recibirSueldo(self, total):
        self._cuentaBancaria.descontarFondos(total)
        self.getCuentaBancaria().anadirFondos(total)

    def __str__(self):
        return (
            "\nNombre: " + self.getNombre() + "\n"
            + "Edad: " + str(self.getEdad()) + "\n"
            + "Cedula: " + str(self.getCedula()) + "\n"
        )
    
    #/*-------------------Getters y setters-------------------------*/
    def getNombre(self):
        return self._nombre

    def setNombre(self, nombre):
        self._nombre = nombre

    def getEdad(self):
        return self._edad

    def setEdad(self, edad):
        self._edad = edad

    def getCedula(self):
        return self._cedula

    def setCedula(self, cedula):
        self._cedula = cedula

    def getCuentaBancaria(self):
        return self._cuentaBancaria

    def setCuentaBancaria(self, cuentaBancaria):
        self._cuentaBancaria = cuentaBancaria

    @classmethod
    def getSalario(cls):
        return cls.SALARIO

    def getTrabajo(self):
        return self._trabajo

    def setTrabajo(self, trabajo):
        self._trabajo = trabajo

    def getIndiceMeta(self):
        return self._indiceMeta

    def setIndiceMeta(self, indiceMeta):
        self._indiceMeta = indiceMeta

    def getVerificadorMetasCumplidas(self):
        return self._verificadorMetasCumplidas

    def setVerificadorMetasCumplidas(self, verificadorMetasCumplidas):
        self._verificadorMetasCumplidas = verificadorMetasCumplidas

    @classmethod
    def getPersonasTotales(cls):
        return cls.personasTotales

    @classmethod
    def getListaPersonas(cls):
        return cls.listaPersonas

File: Java/src/gestorAplicacion/gestion/Persona.java
/*  Autores: Joan Sebastian Salazar Montoya, Monica Sofia Restrepo León, 
Luis Alejandro Varela Ojeda, Maria Fernanda Calle Agudelo, Jaider Castañeda Villa
 * FUNCIONALIDAD DE LA CLASE: clase abstracta que proporciona atributos y métodos 
 * comunes para la creación de los distintos tipos de trabajadores de la empresa.
 */
package gestorAplicacion.gestion;

import java.io.Serializable;
import java.util.ArrayList;


public abstract class Persona implements Serializable {


	/*-------------------Atributos-------------------------*/

	private static final long serialVersionUID = 12736217L;    
	
	
	private String nombre;
	private int edad;
	private int cedula;
	private CuentaBancaria cuentaBancaria;
	private static final int SALARIO=1000;
	private int trabajo;
	private double indiceMeta;
	private ArrayList<Boolean> verificadorMetasCumplidas = new ArrayList<Boolean>();
	private static int personasTotales;
	private static  ArrayList<Persona> listaPersonas = new ArrayList<Persona>();


	
	/*-------------------Constructores-------------------------*/
	
	
	/*Constructor de la clase conductor que recibe todos los parametros*/
	protected Persona(String nombre, int edad, int cedula, CuentaBancaria cuentaBancaria) {
		this.nombre=nombre;
		this.edad=edad;
		this.cedula=cedula;
		this.cuentaBancaria=cuentaBancaria;
		this.trabajo = 0;
		this.indiceMeta = 0;
		this.verificadorMetasCumplidas.add(false);
		this.verificadorMetasCumplidas.add(false);
		listaPersonas.add(this);
		

		}


	/*Constructor de la clase persona que no recibe ninguna parametros*/
	public Persona() {}

	/*------------------- Metodos -------------------------*/
	
	public abstract void recibirSueldo(int total);
	
	
	/*-------------------Getters y setters-------------------------*/

	
	
	public String getNombre() {
		return nombre;
	}
	
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	
	public int getEdad() {
		return edad;
	}
	
	public void setEdad(int edad) {
		this.edad = edad;
	}
	
	public int getCedula() {
		return cedula;
	}
	
	public void setCedula(int cedula) {
		this.cedula = cedula;
	}
	
	public CuentaBancaria getCuentaBancaria() {
		return cuentaBancaria;
	}
	
	public void setCuentaBancaria(CuentaBancaria cuentaBancaria) {
		this.cuentaBancaria = cuentaBancaria;}
	
		public static int getSalario() {
		return SALARIO;
	}
	

	public int getTrabajo() {
		return trabajo;
	}
	
	public void setTrabajo(int trabajo) {
		this.trabajo = trabajo;
	}

	public double getIndiceMeta(){
		return indiceMeta;
	}

	public void setIndiceMeta(double indiceMeta) {
		this.indiceMeta = indiceMeta;
	}

	public ArrayList<Boolean> getVerificadorMetasCumplidas() {
		return verificadorMetasCumplidas;
	}
	
	public void setVerificadorMetasCumplidas(ArrayList<Boolean> verificadorMetasCumplidas) {
		this.verificadorMetasCumplidas = verificadorMetasCumplidas;
	}

	public static int getPersonasTotales() {
		return personasTotales;
	}
	
	public static ArrayList<Persona> getListaPersonas(){
		return listaPersonas;
	}



	

	
}
